debezium-core/src/main/java/io/debezium/pipeline/notification/InitialSnapshotNotificationService.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.pipeline.notification;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import io.debezium.config.CommonConnectorConfig;
+import io.debezium.pipeline.spi.OffsetContext;
+import io.debezium.pipeline.spi.Offsets;
+import io.debezium.pipeline.spi.Partition;
+import io.debezium.pipeline.spi.SnapshotResult;
+import io.debezium.spi.schema.DataCollectionId;
+
+public class InitialSnapshotNotificationService<P extends Partition, O extends OffsetContext> {
```

Comment:
Given that we are going to introduce the blocking snapshot during runtime I think we should come with a different class and aggregate name. It is OK now but we need to think about that.

Reply:
Yes, I agree.